### ISSUE TITLE
historical dates

### ISSUE THREAD

**Issue by simonwibberley** [skeate/Leaflet.timeline#105]
Hi there - thank for the plugin!

I'm working with pre-1970 dates, which seems not to work properly. The slider won't play back or select dates properly, and it gets stuck on 1970-01-01.

Could pre-1970 dates be added easily?

Many thanks

**Comment by simonwibberley** [comment]
Ah, just figured a work around:
 - add 1000 years with moment to the real data
 - subtract 1000 years on the time line slider control formatOutput
moment.js ftw!

**Comment by skorasaurus** [comment]
Perhaps it's something with how your data is formatted? 

I was able to get have historical dates load without any unexpected problems; see https://github.com/skorasaurus/Leaflet.timeline/blob/annex/examples/annex.html for a working example.

**Comment by MarcelWaldvogel** [comment]
I have do display dates in the middle ages. I use a slightly non-linear day number for simplicity  (i.e., there are 4 "February 28" before "March 1"), as nobody will be navigating in single days and we have calendar issues anyway back then. Encoding (in Python):

```python
def epoch(y, m, d):
    # Days since 0001-01-01, given that every month has 31 days
    return ((y-1) * 12 + (m-1)) * 31 + (d-1)
```

Decoding (in JavaScript; yes, using some huge additional libraries would make the code slightly nicer and smaller, but I did not want to have any large dependencies):
```js
formatOutput: function(date) {
  const days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
  var d = date % 31;
  var m = Math.trunc(date / 31) % 12;
  var y = Math.trunc(date / 31 / 12);
  if (d > days[m]) d = days[m];
  d = ("0" + (d+1)).slice(-2);
  m = ("0" + (m+1)).slice(-2);
  y = ("000" + (y+1)).slice(-4);
  return y+"-"+m+"-"+d;
}
```

**Comment by MarcelWaldvogel** [comment]
Of course, even older dates could also be displayed using #97. I had to use the above code, as medieval dates were somehow mapped to 1969-1970 by the JavaScript `Date()` function using the code from [`borders.html`](https://github.com/skeate/Leaflet.timeline/blob/master/docs/examples/borders.html#L65).
